lib/roar/json/json_api/nesting.rb

```diff
@@ -0,0 +1,35 @@
+module Roar
+  module JSON
+    module JSONAPI
+      module Nesting
+        def nested(name, options = {}, &block)
+          options = options.merge(
+            getter:   ->(_opts) { self },
+            setter:   ->(opts)  {},
+            instance: ->(_opts) { self }
+          )
+
+          if block
+            options[:_nested_builder] = nested_builder
+            options[:_base]           = Decorator.default_nested_class
+          end
+
+          property(name, options, &block)
+        end
+
+        NestedBuilder = ->(options) do
+          Class.new(options[:_base]) do
+            feature(*options[:_features])
+            defaults(&DEFAULTS_AS)
```

Comment:
@apotonick Per your comment here https://github.com/trailblazer/roar-jsonapi/issues/13#issuecomment-266230178

> Defaults are not "inherited" into nested representers. I need to think about if that's a good idea or if we should introduce a inline: true option or something.

What do you think about this implementation?